### PR TITLE
feat(freespace_planner): only plan if ego is stopped

### LINKED ISSUE #5936: Controller outputs way too high velocity/acceleration
### Checklist

- [X] I've read the [contribution guidelines](https://github.com/autowarefoundation/autoware/blob/main/CONTRIBUTING.md).
- [X] I've searched other issues and no duplicate issues were found.
- [X] I'm convinced that this is not my fault but a bug.

### Description

I am debugging a freespace planner algorithm with the planning simulator and encountered this crazy situation:

[Screencast from 12-22-2023 01_42_39 PM.webm](https://github.com/autowarefoundation/autoware.universe/assets/18645627/577277db-48ad-4221-b1dc-a59554b7b4e7)

Like astar and other algorithms, the output path of the algorithm is transformed into a trajectory by `freespace_planner` using a fixed speed (default 5km/h):

https://github.com/autowarefoundation/autoware.universe/blob/3a0fd31b95852dedc9a1b26c5496cf31a4e26347/planning/freespace_planner/src/freespace_planner/freespace_planner_node.cpp#L166-L178

The simulation goes usually well, but for some reason this time, the controller started to panic around the goal point and kept outputing very large velocity/acceleration targets. 

I am not sure what triggered that behavior. I just remember the vehicle was stuck nearby the goal pose for a while before (but that's my algorithm.)

### Expected behavior

Velocity/Acceleration should be capped

### Actual behavior

velocity/acceleration diverge

### Steps to reproduce

No idea

### Versions

_No response_

### Possible causes

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by VRichardJP**
Fixes #5936

## Description

See https://github.com/autowarefoundation/autoware.universe/issues/5936#issuecomment-1888475534

## Tests performed

[Screencast from 01-12-2024 03_57_19 PM.webm](https://github.com/autowarefoundation/autoware.universe/assets/18645627/998587b8-d427-4ebc-a929-40803ebb6e36)

## Effects on system behavior

On a parking, freespace planner will only plan new trajectories if the vehicle is stopped. If a new obstacle is found on the current trajectory, the planner will first modify current trajectory to stop the vehicle, and only create a new trajectory once the vehicle is stopped.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/freespace_planner/src/freespace_planner/freespace_planner_node.cpp
Comment (kosuke55) [inline planning/freespace_planner/src/freespace_planner/freespace_planner_node.cpp:445]:
```suggestion
  const bool is_reset_required = !reset_in_progress_ && isPlanRequired();
```

**Review by kosuke55 — APPROVED** [review]
Thanks! LGTM

**Comment by maxime-clem** [comment]
@TakaHoribe, @takayuki5168, can we merge this PR ?

**Comment by takayuki5168** [comment]
@maxime-clem Takeuchi-kun already approved, so I think it's fine.

**Comment by maxime-clem** [comment]
My mistake, I thought the PR was waiting for a code owner to approve, but the `json-schema-check` is what is blocking.
I updated the branch and if the CI passes it should be possible to merge.

**Pull request merged**
